File: lib/enums/transit_mode.dart
enum TransitMode {
  airplane,
  bus,
  cableCar,
  carpool,
  coach,
  ferry,
  funicular,
  gondola,
  monorail,
  rail,
  subway,
  taxi,
  tram,
  trolleybus,
  unknown;

  static TransitMode fromString(String value) {
    switch (value) {
      case 'AIRPLANE':
        return TransitMode.airplane;
      case 'BUS':
        return TransitMode.bus;
      case 'CABLE_CAR':
        return TransitMode.cableCar;
      case 'CARPOOL':
        return TransitMode.carpool;
      case 'COACH':
        return TransitMode.coach;
      case 'FERRY':
        return TransitMode.ferry;
      case 'FUNICULAR':
        return TransitMode.funicular;
      case 'GONDOLA':
        return TransitMode.gondola;
      case 'MONORAIL':
        return TransitMode.monorail;
      case 'RAIL':
        return TransitMode.rail;
      case 'SUBWAY':
        return TransitMode.subway;
      case 'TAXI':
        return TransitMode.taxi;
      case 'TRAM':
        return TransitMode.tram;
      case 'TROLLEYBUS':
        return TransitMode.trolleybus;
      default:
        return TransitMode.unknown;
    }
  }
}

File: src/features/map/domain/models/transit-mode.ts
/*
  To-Do:
    - [ ] Trim the modes Transitous never returns for a stop (car, airplane,
          flex, odm, rental …). They came over from the MOTIS routing API and
          this app only ever renders departures.
*/

export const TransitMode = {
  Walk: 'walk',
  Bike: 'bike',
  Rental: 'rental',
  Car: 'car',
  CarParking: 'carParking',
  CarDropoff: 'carDropoff',
  Odm: 'odm',
  Flex: 'flex',
  Transit: 'transit',
  Tram: 'tram',
  Subway: 'subway',
  Ferry: 'ferry',
  Airplane: 'airplane',
  Suburban: 'suburban',
  Bus: 'bus',
  Coach: 'coach',
  Rail: 'rail',
  HighspeedRail: 'highspeedRail',
  LongDistance: 'longDistance',
  NightRail: 'nightRail',
  RegionalFastRail: 'regionalFastRail',
  RegionalRail: 'regionalRail',
  CableCar: 'cableCar',
  Funicular: 'funicular',
  AerialLift: 'aerialLift',
  ArealLift: 'arealLift',
  Metro: 'metro',
  Other: 'other',
} as const;

// eslint-disable-next-line @typescript-eslint/no-redeclare
export type TransitMode = (typeof TransitMode)[keyof typeof TransitMode];
